src/expression/definitions/within.ts

```diff
@@ -295,12 +295,24 @@ class Within implements Expression {
         if (isValue(args[1])) {
             const geojson = (args[1] as any);
             if (geojson.type === 'FeatureCollection') {
+                const polygonsCoords = [];
                 for (let i = 0; i < geojson.features.length; ++i) {
-                    const type = geojson.features[i].geometry.type;
-                    if (type === 'Polygon' || type === 'MultiPolygon') {
-                        return new Within(geojson, geojson.features[i].geometry);
+                    const {type, coordinates} = geojson.features[i].geometry;
```

Comment:
I think you can change the for i... to be for let polygon of ...
It would help typescript inference.